lib/std/posix.zig

```diff
@@ -102,7 +105,35 @@ pub const STDIN_FILENO = system.STDIN_FILENO;
 pub const STDOUT_FILENO = system.STDOUT_FILENO;
 pub const SYS = system.SYS;
 pub const Sigaction = system.Sigaction;
-pub const Stat = system.Stat;
+pub const Stat = if (no_stat) struct {
+    // The system does not have a native stat structure. Make one up.
```

Comment:
Not OK to make one up. It needs to be `void` instead.

Reply:
Even here in `std.posix`? Isn't the point of this module to present a POSIX layer on top of the system? What I've done here is what both glibc and musl do for `linux-riscv32` since otherwise ~all of userspace would break.

Reply:
Yes, even here. The goals of this API layer are only:
* Abstract over the different ways of returning errors, converting to Zig error sets.
* Abstract over calling `-lc` vs directly making equivalent syscalls.
* Abstract over the *same* APIs across different operating systems.

It is *not* a goal to emulate POSIX APIs when they do not exist. Related: #6600

When one API is trivially expressed in terms of another, it is OK to do, but I think this example steps over that line.